DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/convolution2d/quartus_project_files/software/oneapi_2dconvolution.hpp

```diff
@@ -0,0 +1,153 @@
+#pragma once
+
+// macros for BSP
+#include <system.h>
+// macros for oneAPI kernel
+#include "<path to convolution2d sample>/build/convolution.report.prj/include/register_map_offsets.h"
+// IORD and IOWR
+#include <io.h>
+
+// integer types
+#include <stdint.h>
+
+#define KERNEL_START                        \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_START_REG)
+#define KERNEL_STOP                         \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_STOPCSR_PIPE_CHANNEL_DATA_REG)
+#define KERNEL_STOP_VALID                   \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_STOPCSR_PIPE_CHANNEL_VALID_REG)
+
+#define KERNEL_FINISH_COUNTER               \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_FINISHCOUNTER_REG)
+
+#define KERNEL_STATUS_REGISTER              \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+  ID_CONVOLUTION2D_REGISTER_MAP_STATUS_REG)
+
+#define KERNEL_VERSION                         \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+  ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_VERSIONCSR_PIPE_CHANNEL_DATA_REG)
+
+#define KERNEL_VERSION_VALID                   \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+  ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_VERSIONCSR_PIPE_CHANNEL_VALID_REG)
+
+#define KERNEL_BYPASS                       \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_BYPASSCSR_PIPE_CHANNEL_DATA_REG)
+#define KERNEL_BYPASS_VALID                 \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_BYPASSCSR_PIPE_CHANNEL_VALID_REG)
+
+#define ARG_COEFFS                      \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ARG_ARG_COEFFS_0_REG)
+
+#define ARG_ROWS                            \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ARG_ARG_ROWS_REG)
+
+// add 4 to macro to account for a bug in the header file generated by SYCL HLS
+#define ARG_COLS                            \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ARG_ARG_COLS_REG + 4)
+
+
+namespace oneapiconvolution {
+
+// clang-format off
+#define K_COEFFS_SOBEL_VERTICAL        \
+  {                                    \
+    -1.0f / 6.0f, 0.0f, 1.0f / 6.0f,   \
+    -1.0f / 6.0f, 0.0f, 1.0f / 6.0f,   \
+    -1.0f / 6.0f, 0.0f, 1.0f / 6.0f    \
+  }
+
+#define K_COEFFS_IDENTITY   \
+  {                         \
+    0.0f, 0.0f, 0.0f,       \
+    0.0f, 1.0f, 0.0f,       \
+    0.0f, 0.0f, 0.0f        \
+  }
+
+void start() {
```

Comment:
The naming of these functions seems inconsistent with the other files.

Reply:
these are embedded c code and follow a different standard. Although, I see that I didn't consistently follow that standard, so i've updated it :)